### ISSUE TITLE
Encoding Public Key (.pem)

### ISSUE THREAD

**Issue by biggledfred** [invisal/god_crypto#7]
I'm pretty new to all this, so apologies for any silly questions.  

So you can parse a key like this:

```
import { RSA } from "https://deno.land/x/god_crypto/rsa.ts";

// Parsing public/private key
const publicKey = RSA.parseKey(Deno.readTextFileSync("./public.pem"));
```

Can you go the other way?
I.e. take an RSA object and generate/encode the pem.

**Comment by invisal** [comment]
So basically, you want to create PEM format from the parsed key? Do you have recommended syntax? I can take a look and probably add this implementation as soon as possible.

**Comment by biggledfred** [comment]
Yes absolutely, but firstly thanks again for your good work with this library, it's awesome.   

So my specific use case is for JWT verification for cognito ([this is a great outline of the process](https://itnext.io/aws-cognito-example-using-react-ui-and-node-js-rest-apis-part-3-jwt-secured-rest-apis-e56d336ce306)).  I'm trying to create a clone of this nodeJS module called [ExpressCognito ](https://github.com/ghdna/cognito-express/blob/master/lib/strategy.js) but for Deno and Oak.

With cognito we get JSON web keys in [this format](https://cognito-idp.eu-west-2.amazonaws.com/eu-west-2_z1Go5XdrZ/.well-known/jwks.json):
```
{  
  "keys":[  
    {  
      "alg":"RS256",
      "e":"AQAB",
      "kid":"0n7zsh4x4eg5BFk3fOo6SKyi0zrG83DK+2C/5xmFqM0=",
      "kty":"RSA",
      "n":"tKQOwFsFCRSNiL18QN77UaYYQvBvgIr-ImRW-9Y6KnfYeS5Tk-O39JjllVCYNf6EB5xcZELQuaJmh_bm8uiOxFuk1t6pCghyQWUWU_uR8BRhrqxF-ugWbnWWfkGqPjMD9Tc3oGVIZRwqpXSXkGaid6DaqhkQIzrYQtPwUMge4w4oQgl5KuowBxrxjJdrRChi4FPABNoRaB8VJOHAgvoobP9VACRkooSXKz5b0aCzExuv5mkg8WnhJi_xQCjX41QX3RbpwvgoqPrCxYmCjngy5qwItsJSOozLdxwFp6EDQExp5VwRfd0xU112Ny0ea-EeDlgkQMgVLfvb7cOm74f5WQ",
      "use":"sig"
    },
    {  
      "alg":"RS256",
      "e":"AQAB",
      "kid":"iPeYlnLRSbCoyi+JdywuePKNdluh4+7tIPaOn8yWZuE=",
      "kty":"RSA",
      "n":"iPu6YsSrKbVMMMmhwcyQjdzJHY9y53nEj-oHF3VimQu8gZ0qXaQRckJudZ0AOzDHZvSfdP2mLUMa6HWa_1n7NNsBWCRWTjqOXex2iAOX2Ryo9Sa_pRmXOCEPAbQor2YdYQdKqetbUllhPXmzTRtCGEcXbn8bB40rhAAcRoFUhBqWyUxIQPzwZlhTzk41u5E0V3iIt2jnFTTXfxgHj2571VTCHYTyqdOqcmdx4zvVaY8SUEg-VFRy4GM76JBEgttv7AhPtRlGkhKgIpPb8UJPDXMdtfFaHPzWOGGe9qh1YsuCeEqIUCwTmhy4sO8yVn6ylhJZpFAF8zQLtxIvYRwzdw",
      "use":"sig"
    }
  ]
}
```

I want to convert these keys into PEM, so that I can then verify it using the Deno [djwt module](https://deno.land/x/djwt@v1.2) or something similar.  I'm assuming the key in the djwt library is the PEM container, but I'll burn that bridge when I get to it.

So we can create the instantiate the RSA object with the above JSON as such:
`const rsa = new RSA(RSA.importKey(<JWKs from cognito>));`

then perhaps encode it / export it like this:

`const pem = rsa.exportKey("pem");` perhaps the exportKey method could have optional arguments to allow for different exports 

I've noticed RSA.parseKey and RSA.importKey do the same thing is there any reason for having both?

**Comment by invisal** [comment]
When I first started the library, I created function `parseKey`, but then I want it to be a bit consistent with WebCrypto naming so I decide to change to `importKey`. Anyway, it should be easy to do this. There are two syntax that I can think of

**Syntax 01**
```
const pem = RSA.importKey(<JWK>).to("pkcs8");
```

or

**Syntax 02**
```
const pem = RSA.exportKey(RSA.importKey(<JWK>), "pkcs8");
```

I will try to work on this feature during this week.

**Comment by biggledfred** [comment]
The syntax 01 is really neat.  

I think the .exportKey (syntax 02) would be a nice **non-static method** to have once you instantiated RSA instance with your key.  But that would be easy enough to do once the change above is there.  I'd be happy to implement it and create a PR for that later, if that is something you'd like done?

**Comment by invisal** [comment]
I am already working on it now. It should be done by tomorrow.

**Comment by invisal** [comment]
Featured is added. I have released it on version `v1.4.2`.

**Comment by biggledfred** [comment]
![](https://media0.giphy.com/media/WV9Xx9itqEIu8GQloO/giphy.gif)
